File: frontend/src/app/models/leave.ts
export interface LeaveDto {
  leaveId: number;
  employeeId: number;
  employeeName: string | null;
  fromDate: string;
  toDate: string;
  leaveType: string;
  reason: string;
  status: string;
  approvedByName: string | null;
  appliedAt: string;
  reviewedAt: string | null;
}

export interface ApplyLeaveDto {
  fromDate: string;
  toDate: string;
  leaveType: string;
  reason: string;
}

export const LEAVE_TYPES = ['Sick', 'Casual', 'Earned'] as const;
export const LEAVE_STATUSES = ['Pending', 'Approved', 'Rejected'] as const;

File: backend/WMS.Domain/Entities/Leave.cs
using System.ComponentModel.DataAnnotations;

namespace WMS.Domain.Entities
{
    public class Leave : BaseEntity
    {
        public int LeaveId { get; set; }
        public int EmployeeId { get; set; }
        public Employee Employee { get; set; } = null!;
        [Required]
        public DateOnly FromDate { get; set; }
        [Required]
        public DateOnly ToDate { get; set; }
        [Required, MaxLength(20)]
        public string LeaveType { get; set; } = "Casual";
        [MaxLength(1000)]
        public string? Reason { get; set; }
        [Required, MaxLength(20)]
        public string Status { get; set; } = "Pending";
        public int? ApprovedBy { get; set; }
        public Employee? Approver { get; set; }
        public DateTime AppliedAt { get; set; } = DateTime.UtcNow;
        public DateTime? ReviewedAt { get; set; }
    }
}
